### PR TITLE
Change the required Terraform version from 0.9.10 to 0.10

### PR DESCRIPTION
**Pull request by deanwilson**
We're attempting to stay current.

### CODE REVIEW

**Comment by issyl0** [comment]
I think this can come out of WIP now - I fixed the pre-commit `terraform validate` hook.

File: tools/terraform-validate.sh [hunk 1]
```diff
@@ -6,5 +6,6 @@ for file in "$@"; do
 done \
   | sort | uniq | \
     while read -r dir; do
+      terraform init "${dir}";
```
Comment (surminus) [inline tools/terraform-validate.sh]: We should purge any previous terraform config, so:
```
rm -rf ${dir}/.terraform
rm -f ${dir}/terraform.tfstate.backup
```

Comment (surminus) [inline tools/terraform-validate.sh]: Also, does an init require exporting of credentials?

If so, then this may break Jenkins tests.

Comment (issyl0) [inline tools/terraform-validate.sh]: As discussed, I removed the pre-commit hook trigger from the Jenkinsfile (context in commit message).

**Comment by surminus** [comment]
I wonder if we should now update to 0.10.4 🤔

**Comment by issyl0** [comment]
Did we decide what to do with this?

**Comment by surminus** [comment]
We should push to the newest version and test.

**Comment by issyl0** [comment]
I've updated to `0.10.6` and included the new provider versions for Fastly and AWS. Tested with a `plan` on a couple of projects, and things seem to work. This is ready for a re-review.

**Comment by issyl0** [comment]
I've updated to `0.10.6` and included the new provider versions for Fastly and AWS. Tested with a `plan` on a couple of projects, and things seem to work. This is ready for a re-review.

**Comment by surminus** [comment]
I would almost be tempted to set up some tests for these that check if one module is out of sync with the others. 

I've seen a fair few copy-pasta fails, and a test seems like a good thing here.

**Comment by issyl0** [comment]
@surminus I added a test for the Terraform version. Would you like providers too?

**Comment by surminus** [comment]
Great, that seems to do the job! I already got some failures.

Would be good to add to the Jenkinsfile or pre-commit hooks. 

I'm still a little wary of not having `terraform validate` being put into CI, but I think the need to upgrade to a later version is greater.

**Comment by surminus** [comment]
@issyl0 if you're happy to rebase and add your tests to CI or pre-commit, then I'd be happy to approve and merge before we get even more out of date

**Comment by issyl0** [comment]
We've got 0.10.7 now, so I'll do this dance again and rebase and then we can merge.

**Pull request merged**